### PR TITLE
Implement delete for all core plugin entries except the volume package

### PR DESCRIPTION
**Pull request by ekinanp**
The volume package will be a bit more involved, so that work will be done separately.

### CODE REVIEW

File: plugin/aws/s3Bucket.go [hunk 1]
```diff
@@ -186,6 +196,21 @@ func (b *s3Bucket) List(ctx context.Context) ([]plugin.Entry, error) {
 	return listObjects(ctx, b.client, b.Name(), "")
 }
 
+func (b *s3Bucket) Delete(ctx context.Context) (bool, error) {
+	// According to https://docs.aws.amazon.com/AmazonS3/latest/dev/delete-or-empty-bucket.html,
+	// we must delete the bucket's objects and object versions (for versioned buckets) before
+	// deleting the bucket itself. We defer deleting object versions as part of supporting
+	// versioned buckets, which is a separate piece of work.
```
Comment (ekinanp) [inline plugin/aws/s3Bucket.go]: Should this be an issue, or should we wait until someone requests versioned bucket support?

**Comment by ekinanp** [comment]
Any activity that'd be useful to log here? The implementations seem simple enough that I figured there was no need.

File: plugin/aws/s3Bucket.go (hunk 1 above)
Comment (MikaelSmith) [reply plugin/aws/s3Bucket.go]: We should create an issue, but probably won't start on it yet.

File: plugin/docker/container.go [hunk 2]
```diff
@@ -88,6 +88,14 @@ func (c *container) List(ctx context.Context) ([]plugin.Entry, error) {
 	return []plugin.Entry{clf, cm, vol.NewFS("fs", c, 3)}, nil
 }
 
+func (c *container) Delete(ctx context.Context) (bool, error) {
+	err := c.client.ContainerRemove(ctx, c.id, types.ContainerRemoveOptions{
+		// TODO: Should we set RemoveVolumes and RemoveLinks too?
```
Comment (MikaelSmith) [inline plugin/docker/container.go]: I'm not sure I understand what these mean. I'd guess we don't want to. I think those are things that can persist after and be reused, so they should be removed separately.

File: plugin/kubernetes/pod.go [hunk 3]
```diff
@@ -65,3 +66,7 @@ func (p *pod) List(ctx context.Context) ([]plugin.Entry, error) {
 	return entries, nil
 }
 
+func (p *pod) Delete(ctx context.Context) (bool, error) {
+	err := p.client.CoreV1().Pods(p.ns).Delete(p.Name(), &v1.DeleteOptions{})
```
Comment (MikaelSmith) [inline plugin/kubernetes/pod.go]: `v1` is already imported as `metav1`.

File: plugin/kubernetes/pvc.go [hunk 4]
```diff
@@ -61,6 +62,11 @@ func (v *pvc) List(ctx context.Context) ([]plugin.Entry, error) {
 	return volume.List(ctx, v)
 }
 
+func (v *pvc) Delete(ctx context.Context) (bool, error) {
+	err := v.pvci.Delete(v.Name(), &v1.DeleteOptions{})
```
Comment (MikaelSmith) [inline plugin/kubernetes/pvc.go]: `v1` is already imported as `metav1`.

**Review by MikaelSmith — CHANGES_REQUESTED** [review]

File: plugin/docker/container.go (hunk 2 above)
Comment (ekinanp) [reply plugin/docker/container.go]: I noticed those options in `docker rm` so thought I'd ask. I'll remove the TODO.

**Comment by ekinanp** [comment]
Updated to address the feedback. Also filed https://github.com/puppetlabs/wash/issues/561 to track versioned buckets.

**Review by MikaelSmith — APPROVED** [review]

**Comment by MikaelSmith** [comment]
Supports #479.

**Pull request merged**